rq/worker.py

```diff
@@ -839,7 +844,7 @@ def stop_scheduler(self):
                 pass
             self.scheduler._process.join()
 
-    def dequeue_job_and_maintain_ttl(self, timeout: int) -> Tuple['Job', 'Queue']:
+    def dequeue_job_and_maintain_ttl(self, timeout: Optional[int], max_idle_time: Optional[int] = None) -> Tuple['Job', 'Queue']:
```

Comment:
Shouldn't `max_idle_time` be factored into `timeout` calculation? I feel like there should be something like this:
```python
dequeue_timeout = min(max_idle_time, timeout)
self.queue_class.dequeue_any(.., timeout)
```

Reply:
I agree, I changed it a bit to really be closer to `max` in `max_idle_time` - wait less in case there is only a few more seconds left until we reach `max_idle_time`, LMKWYT